integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/deleteTable/TestDeleteTableNewDDL.scala

```diff
@@ -58,6 +67,20 @@ class TestDeleteTableNewDDL extends QueryTest with BeforeAndAfterAll {
     }.getMessage.contains("Database 'dropdb_test' not found"))
   }
 
+  test("test drop database when dblocation is inconsistent") {
+    var dbName = "dropdb_test"
+    sql(s"drop database if exists $dbName cascade")
+    CarbonProperties.getInstance().addProperty(CarbonCommonConstants.STORE_LOCATION,
+      warehouse + File.separator + "carbonwarehouse")
+    sql(s"create database $dbName")
```

Comment:
I think, create database itself should not be allowed

Reply:
if users follow these steps, problems maybe happen:
step1. create database db, under carbon.storelocation = spark.sql.warehouse.dir = /somelocation
step2. modify carbon.storelocation = /anotherlocation
step3. drop database db

Both /somelocaton/db and /anotherlocation/db will be cleared.

Yeah, you are right, under different configuraions, both CREATE DATABASE and DROP DATABASE shall not be allowed.

Is that right?

Reply:
@QiangCai  what's your opinion?

Reply:
> I think, create database itself should not be allowed. Since database is already created, location is registered to hive. So, this should be handled while create DB

I agree. Better to create a database in `spark.sql.warehouse.dir` only as` carbon.storelocation` is deprecated.

Reply:
carbon.storeLocation is not deprecated right now.

I suggest the following changes
1. Deprecate the use of carbon.storeLocation(this property should be removed in 3.0 release)
2. throw exception on database creation and deletion as suggested by indhumathi.
3. Add a documentation point that if the user wants to delete the database then store has to be migrated to "spark.sql.warehouse.dir"

Reply:
in this testcase, you can handle throw exception, on create db itslef

Reply:
i have modified code accoring to your suggestions.
Now exception will be thrown on database creation and deletion.
add the note to let users to remove ''carbon.storelocaton'